src/java/org/apache/cassandra/cql3/Relation.java

```diff
@@ -174,14 +175,15 @@ public final boolean isSlice()
      *
      * @param table the Column Family meta data
      * @param boundNames the variables specification where to collect the bind variables
+     * @param indexHints the query index hints, used to disambiguate relations that might be supported by multiple indexes
      * @return the <code>Restriction</code> corresponding to this <code>Relation</code>
      * @throws InvalidRequestException if this <code>Relation</code> is not valid
      */
-    public final Restriction toRestriction(TableMetadata table, VariableSpecifications boundNames)
+    public final Restriction toRestriction(TableMetadata table, VariableSpecifications boundNames, IndexHints indexHints)
     {
         switch (relationType)
         {
-            case EQ: return newEQRestriction(table, boundNames);
+            case EQ: return newEQRestriction(table, boundNames, indexHints);
```

Comment:
Does this mean that index hints are only used for equality conditions? SAI can be used for ranges too. Or do I misunderstand the meaning of this?